cumulus/primitives/core/src/lib.rs

```diff
@@ -78,6 +78,43 @@ impl From<MessageSendError> for &'static str {
 	}
 }
 
+/// The origin of an inbound message.
+#[derive(Encode, Decode, MaxEncodedLen, Clone, Eq, PartialEq, TypeInfo, Debug)]
+pub enum AggregateMessageOrigin {
+	/// The message came from the para-chain itself.
+	Loopback,
```

Comment:
Small nit: Maybe to be consistent with the naming convention in XCM MultiLocations, where the `Here` junction means _self_.

```suggestion
	Here,
```